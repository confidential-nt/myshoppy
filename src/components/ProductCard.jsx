import React from "react";

export default function ProductCard({ product }) {
  console.log(product);
  return (
    <li className="basis-1/4 mr-3 last:mr-0 cursor-pointer shadow-lg hover:scale-105 transition">
      <div className="w-full">
        <img
          src={product.imageURL}
          alt={product.name}
          className="w-full h-full"
        />
      </div>
      <div className="p-4">
        <div className="flex justify-between">
          <h4>{product.name}</h4>
          <strong className="font-normal">â‚©{product.price}</strong>
        </div>
        <span className="text-sm text-gray-700">{product.category}</span>
      </div>
    </li>
  );
}
