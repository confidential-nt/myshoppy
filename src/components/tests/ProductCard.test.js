import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { withRouter } from "../../tests/utils";
import { Route, useLocation } from "react-router-dom";
import ProductCard from "../ProductCard";
import { fakeProduct } from "../../tests/products";

describe("ProductCard", () => {
  it("스냅샷 테스트", () => {
    const { asFragment } = renderProductCard();
    expect(asFragment()).toMatchSnapshot();
  });

  it("올바른 제품 정보를 보여줘야한다", () => {
    renderProductCard();
    expect(screen.getByText(fakeProduct.name)).toBeInTheDocument();
    expect(screen.getByText(fakeProduct.category)).toBeInTheDocument();
    expect(screen.getByText(`₩${fakeProduct.price}`)).toBeInTheDocument();
  });

  it("제품 카드를 클릭했을 때 제품 디테일로 이동함과 동시에 올바른 제품 카드 정보가 들어와야한다", async () => {
    renderProductCard();
    const card = screen.getByRole("listitem");
    await userEvent.click(card);

    expect(
      screen.getByText(JSON.stringify({ product: fakeProduct }))
    ).toBeInTheDocument();
  });

  function renderProductCard() {
    return render(
      withRouter(
        <>
          <Route
            element={<ProductCard id={1} product={fakeProduct} />}
            path="/"
          />
          <Route element={<ProductDetail />} path={"/products/1"} />
        </>
      )
    );
  }

  function ProductDetail() {
    return <p>{JSON.stringify(useLocation().state)}</p>;
  }
});
