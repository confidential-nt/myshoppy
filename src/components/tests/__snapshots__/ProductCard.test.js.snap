// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard 스냅샷 테스트 1`] = `
<DocumentFragment>
  <li
    class="cursor-pointer shadow-lg hover:scale-105 transition"
  >
    <div
      class="w-full"
    >
      <img
        alt="핫핑크 후드 HOODY"
        class="w-full h-full"
        src="http://image/"
      />
    </div>
    <div
      class="p-4"
    >
      <div
        class="flex justify-between"
      >
        <h4>
          핫핑크 후드 HOODY
        </h4>
        <strong
          class="font-normal"
        >
          ₩40000
        </strong>
      </div>
      <span
        class="text-sm text-gray-700"
      >
        여성
      </span>
    </div>
  </li>
</DocumentFragment>
`;
